### PR TITLE
feat: Add Q&A and Summarization Prompt Templates

### PR DESCRIPTION
**Pull request by nate-mar**
Relates to #1305 

This PR adds the prompt templates for Q&A and summarization use cases. 

It also adds an llm output map as a convenience for setting rails on the evals call, as well as for mapping golden dataset values.

### CODE REVIEW

**Comment by nate-mar** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by nate-mar** [comment]
recheck

File: src/phoenix/experimental/evals/templates/default_templates.py [hunk 1]
```diff
@@ -38,3 +38,46 @@
     # Answer: {response}
     Is the answer above factual or hallucinated based on the query and reference text?
 """
+
+Q_AND_A_PROMPT_TEMPLATE_STR = """
```
Comment (axiomofjoy) [inline src/phoenix/experimental/evals/templates/default_templates.py:42]: Nit: `QA_PROMPT_TEMPLATE_STR`

File: src/phoenix/experimental/evals/templates/default_templates.py [hunk 2]
```diff
@@ -38,3 +38,46 @@
...
+Your response must be a single word, either "correct" or "incorrect",
+and should not contain any text or characters aside from that word.
+"correct" means that the question is correctly and fully answered by the answer.
+"incorrect" means that the question is not correctly or only partially answered by the
+answer.
+"""
+# The prompt output map is used to map 1) to provide rails to the llm in order to constrain
+# the llm's outputs to the expected values. 2) golden dataset ground truth boolean values
+# to the llm output
+Q_AND_A_PROMPT_OUTPUT_MAP = {True: "correct", False: "incorrect"}
```
Comment (axiomofjoy) [inline src/phoenix/experimental/evals/templates/default_templates.py:62]: We added a new convention above for rails for hallucinations and relevancy.

**Review by axiomofjoy — APPROVED** [review]

**Comment by jlopatec** [comment]
We added the approach above of a map, so there was a clear map for rails and benchmark data sets. The move to only a list means that we still need to manually put in the map for the benchmark data sets. 

Q_AND_A_PROMPT_OUTPUT_MAP = {True: "correct", False: "incorrect"}

-- The values of the map are used for the rails -- 

Q_and_A_classifications = llm_eval_binary(
    dataframe=df_sample,
    template=Q_AND_A_PROMPT_TEMPLATE_STR,
    model=model,
    rails=Q_AND_A_PROMPT_TEMPLATE_MAP.values(),
)

-- I need the map for the benchmark dataset. I'm not sure how you are going to do the following without manually looking at the template and building a map? I shouldn't have to look inside the template to use it.

true_labels = df_sample["answer_true"].map(**Q_AND_A_PROMPT_TEMPLATE_MAP**).tolist()
predicted_labels = Q_and_A_classifications

print(classification_report(true_labels, predicted_labels, labels=classes))
confusion_matrix = ConfusionMatrix(
    actual_vector=true_labels, predict_vector=predicted_labels, classes=**Q_AND_A_PROMPT_TEMPLATE_MAP.values()**
)

**Comment by mikeldking** [comment]
> We added the approach above of a map, so there was a clear map for rails and benchmark data sets. The move to only a list means that we still need to manually put in the map for the benchmark data sets.
> 
> Q_AND_A_PROMPT_OUTPUT_MAP = {True: "correct", False: "incorrect"}
> 
> -- The values of the map are used for the rails --
> 
> Q_and_A_classifications = llm_eval_binary( dataframe=df_sample, template=Q_AND_A_PROMPT_TEMPLATE_STR, model=model, rails=**Q_AND_A_PROMPT_OUTPUT_MAP.values()**, )
> 
> -- I need the map for the benchmark dataset. I'm not sure how you are going to do the following without manually looking at the template and building a map? I shouldn't have to look inside the template to use it, the builder of the template is the one who knows the map, not the user of the template.
> 
> true_labels = **df_benchmark**["answer_true"].map(**Q_AND_A_PROMPT_OUTPUT_MAP**).tolist() predicted_labels = Q_and_A_classifications
> 
> print(classification_report(true_labels, predicted_labels, labels=classes)) confusion_matrix = ConfusionMatrix( actual_vector=true_labels, predict_vector=predicted_labels, classes=**Q_AND_A_PROMPT_OUTPUT_MAP.values()** )

The convention is there, it's just called RAILS instead of map which doesn't match the parameter name

**Pull request merged**